### ISSUE TITLE
What would be the preferred way to mount linprocfs in a pot jail?

### ISSUE THREAD

**Issue by 0mp** [pizzamig/pot#87]
I'd like to run a Linux binary with Linuxulator in a pot jail and I need the Linux filesystems like linprocfs to be available in the pot. 

What would be the preferred way to mount linprocfs in a pot jail? Is it out of the scope of the pot framework and should be handled manually by the admin?

**Comment by pizzamig** [comment]
no, it's not out of scope, even if from inside you cannot load the Linuxulator module.
It could be possible to implement an attribute for those features, similarly to `procfs` and `fdescfs`

While waiting for those attributes to be implemented, there is a way to do everything you need automatically using hooks (feature not properly documented yet, but available).
Hooks are script that can be executed before/after start/stop in the host environment, outside the jail, that you can set using the `set-hook` command
Hooks are the ideal way to extend/customize the framework.

To proper support linuxulator in a jail, I guess all Linux related feature should be supported, like
* linprocfs
* linsysfs
* linux.osname, linux.osrelease, linux.oss_version

Am I missing something?

**Comment by 0mp** [comment]
> no, it's not out of scope, even if from inside you cannot load the Linuxulator module.
> It could be possible to implement an attribute for those features, similarly to `procfs` and `fdescfs`

Nice, that's good news.


> While waiting for those attributes to be implemented, there is a way to do everything you need automatically using hooks (feature not properly documented yet, but available).

I've got one problem with hooks, however. How do I know from a hook script what is the path to my jail? So for example if I want to mount linprocfs in a jail, then I've got to know its "mount" path.

> 
> To proper support linuxulator in a jail, I guess all Linux related feature should be supported, like
> 
>     * linprocfs
> 
>     * linsysfs
> 
>     * linux.osname, linux.osrelease, linux.oss_version
> 
> 
> Am I missing something?

I am not sure, I've not played around with Linuxulator that much in a jail context. I just mounted desired filesystems manually to get the Folding@home client working (and it just worked!):

```
# fstab
linprocfs   /opt/pot/jails/fah/m/compat/linux/proc  linprocfs       rw,failok,noauto      0       0
linsysfs    /opt/pot/jails/fah/m/compat/linux/sys   linsysfs        rw,failok,noauto      0       0
tmpfs       /opt/pot/jails/fah/m/compat/linux/dev/shm  tmpfs   rw,mode=1777,failok,noauto       0       0
```
